app/main/helpers/framework_helpers.py

```diff
@@ -16,6 +18,15 @@ def get_latest_live_framework(all_frameworks, framework_type):
         return None
 
 
+def get_latest_live_framework_or_404(all_frameworks, framework_type):
+    latest_live_framework = get_latest_live_framework(all_frameworks, framework_type)
+
+    if not latest_live_framework:
+        abort(404, f"No latest framework found for `{framework_type}` family")
```

Comment:
I'd have thought that if this ever actually happens, something has gone terribly wrong, so surely dying loudly would be better than 404'ing? If this really can happen then of course `get..or_404` will avoid unnecessary noise in the logs and that's fine.

Reply:
Mm, it shouldn't happen under normal circumstances. But I think it would be annoying (and weirdly fragile) to generate a 500 if a user changed `...gov.ul/digital-outcomes-and-specialists/opportunities` to `...gov.uk/blah/opportunities` (or whatever).

Reply:
We should probably rename these to `framework_family` whilst we're here (or in the other PR).

Reply:
I'm going to do separate PRs for those in the rest of the apps soon.

Reply:
Yeah, done as part of the rebase.

Reply:
I _wasn't_ asking you to go around all the apps doing this everywhere, unless you happen to be looking at that code anyway :-D

Reply:
It's an easy enough change and I'd like to just get it over with. 👍 

Actually changing the API responses will be harder. I need to give it some thought.